services/device-registry-base/src/main/java/org/eclipse/hono/service/management/device/DeviceManagementService.java

```diff
@@ -102,6 +102,7 @@ public interface DeviceManagementService {
      *             Implementations <em>must not</em> invoke the {@link Span#finish()} nor the {@link Span#finish(long)}
      *             methods. However,implementations may log (error) events on this span, set tags and use this span
      *             as the parent for additional spans created as part of this method's execution.
+     * @param isGateway Optional filter for searching all gateways or only devices.
```

Comment:
can you please make it more explicit what will be returned if an empty optional is passed in?

Reply:
Done